### PR TITLE
Add georeference to OSM file

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

Opendrive2lanelet2convertor.py now writes geoference data into the .osm file when converting it. 

## Motivation and Context

This enables the map_param_loader and lanelet2_map_laoding nodes to read the georeference directly without relying on external config files. Previously, the georeference file was baked into the docker image when it was using config files for parameters.

## How Has This Been Tested?

Converted a sample .xodr file into .osm. and have the geoference info parsed by map_loading_param and lanelet2_map_laoding nodes in map_file package in autoware successfully.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: engineering_tools/opendrive_lanelet2_converter/opendrive2lanelet2convertor.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ def __init__(self, fn):
         self.relations = xml.Element('relations')
         self.all_nodes = []
         self.all_ways = []
+        self.projector_type = '1'
+        self.base_frame = '+proj=geocent +ellps=WGS84 +datum=WGS84 +units=m +no_defs'
+        self.target_frame = '+proj=tmerc +lat_0=38.95197911150576 +lon_0=-77.14835128349988 +k=1 +x_0=0 +y_0=0 +units=m +vunits=m'
```
Comment (msmcconnell) [inline engineering_tools/opendrive_lanelet2_converter/opendrive2lanelet2convertor.py]: This target_frame should be read in from the xml file. I think you could just use self.geoReference as the target_frame.

**Comment by msmcconnell** [comment]
Relates to
#557

File: engineering_tools/opendrive_lanelet2_converter/opendrive2lanelet2convertor.py [hunk 2]
```diff
@@ -1,283 +0,0 @@
...
-    # convert x,y values to geo points using the geo reference defined in the input file
-    def get_point_geo(self,x,y):
-        inProj = Proj(self.geoReference)
-        return inProj(x,y,inverse=True)
-
-    def write_xml_to_file(self,fn):
-                
-        geoRefXml = xml.Element('geoReference')
-        geoRefXml.text = self.geoReference
-        self.root.append(geoRefXml)
```
Comment (MishkaMN) [inline engineering_tools/opendrive_lanelet2_converter/opendrive2lanelet2convertor.py]: This is the line needed to write geoRereference into osm file while converting. 
The file is entirely deleted in this branch to have minimum conflict with develop.
However, this script will be added into separate library in the near future.

**Comment by Supraluminal** [comment]
@zijianzhang The `openlanelet2converter.py` file has been migrated to our fork of opendrive2lanelet here: https://github.com/usdot-fhwa-stol/opendrive2lanelet. The file you mention specifically can be found here: https://github.com/usdot-fhwa-stol/opendrive2lanelet/blob/develop/opendrive2lanelet2convertor.py

**Pull request merged**